src/utils/file/file_utils.cpp

```diff
@@ -0,0 +1,247 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <errno.h>
+#include <file_utils.h>
+#include <filesystem>
+#include <algorithm>
+#include <fcntl.h>
+#include "common/nixl_log.h"
+#include <absl/strings/str_format.h>
+
+int nixlFileUtils::getModeFlags(nixlFileMode mode, bool use_direct) {
+    // Base flags for each mode
+    switch (mode) {
+        case nixlFileMode::CREATE:
+        case nixlFileMode::CREATE_DIRECT:
+            return ((mode == nixlFileMode::CREATE_DIRECT || use_direct) ? O_DIRECT : 0) |
+                   O_CREAT | O_WRONLY | O_TRUNC;
+
+        case nixlFileMode::READ_ONLY:
+        case nixlFileMode::READ_DIRECT:
+            return ((mode == nixlFileMode::READ_DIRECT || use_direct) ? O_DIRECT : 0) |
+                   O_RDONLY;
+
+        case nixlFileMode::WRITE_ONLY:
+        case nixlFileMode::WRITE_DIRECT:
+            return ((mode == nixlFileMode::WRITE_DIRECT || use_direct) ? O_DIRECT : 0) |
+                   O_WRONLY;
+
+        case nixlFileMode::READ_WRITE:
+        case nixlFileMode::READ_WRITE_DIRECT:
+            return ((mode == nixlFileMode::READ_WRITE_DIRECT || use_direct) ? O_DIRECT : 0) |
+                   O_RDWR | O_CREAT;
+
+        default:
+            return -1;
+    }
+}
+
+nixl_status_t nixlFileUtils::openFile(const std::string& path, nixlFileMode mode, int& fd, bool use_direct) {
+    int flags = getModeFlags(mode, use_direct);
```

Comment:
same here. what is the reason to define NIXL flags, error codes, etc for filesystem access?
maybe just need specific things that don't exist such as prefixedPath?
it adds more code we need to maintain, review, and potentially fix issues

Reply:
We need to handle filesystem open/close for the following reasons:
1. To Enable opening/closing files from Plugins (during register/der-register)
2. To support passthorugh such that fds are opened without compatibility issues between RUST/Python/C++
3. To facilitate ease of testing

Reply:
you mean to say you don't want NIXL error codes? Since its wrapped this was done for consistency and ease of use in NIXL plugins.